thirdparty/hnswlib/hnswlib/hnswalg.h

```diff
@@ -600,24 +610,29 @@ class HierarchicalNSW : public AlgorithmInterface<dist_t> {
     }
 
     void
-    loadIndex(const std::string& location, SpaceInterface<dist_t>* s, size_t max_elements_i = 0) {
-        std::ifstream input(location, std::ios::binary);
-
-        if (!input.is_open())
-            throw std::runtime_error("Cannot open file");
+    loadIndex(const std::string_view location, const knowhere::LoadConfig& config, size_t max_elements_i = 0) {
+        auto input = knowhere::FileReader(location, true);
```

Comment:
Why we need a new file stream here?

Reply:
We won't map on the entire file, only the data part, or result in a poor performance